File: Hunt_Career_Cypress/cypress/pageObjects/RegisterPage.js
import { BasePage } from "./BasePage";
import LoginPage from "./LoginPage";

export default class RegisterPage extends BasePage {
    getFirstNameField() {
        return cy.xpath("//input[@id='firstName']");
    }
    getLastNameField() {
        return cy.xpath("//input[@id='lastName']");
    }
    getEmailField() {
        return cy.xpath("//input[@id='email']");
    }
    getPasswordField() {
        return cy.xpath("//input[@id='password']");
    }
    getConfirmPasswordField() {
        return cy.xpath("//input[@id='confirmPassword']");
    }
    getPhoneNumberField() {
        return cy.xpath("//input[@id='phoneNumber']");
    }
    getFirstNextButton() {
        return cy.xpath("//button[@type='button']");
    }
    getSecondNextButton() {
        return cy.xpath("//button[normalize-space()='Next']");
    }
    getSignupButton() {
        return cy.xpath("//button[@aria-label='Sign up']");
    }
    getDuplicateErrorMessage() {
        return cy.xpath("//div[contains(text(),'User already exists')]");
    }
    getFirstNameBlankMessage() {
        return cy.xpath("//div[contains(text(),'First Name is required.')]");
    }
    getLastNameBlankMessage() {
        return cy.xpath("//div[contains(text(),'Last Name is required.')]");
    }
    getInvalidEmailAddressMessage() {
        return cy.xpath("//div[contains(text(), 'Please enter a valid email address.')]");
    }
    getEmailBlankMessage() {
        return cy.xpath("//div[contains(text(), 'Email is required.')]");
    }
    getPasswordLengthErrorMessage() {
        return cy.xpath("//div[contains(text(), 'Password must be at least 8 characters long.')]");
    }
    getPasswordMismatchMessage() {
        return cy.xpath("//div[contains(text(), 'Passwords do not match.')]");
    }
    getPhoneNumberRequiredMessage() {
        return cy.xpath("//div[contains(text(), 'Phone Number is required.')]");
    }
    getInvalidPhoneNumberMessage() {
        return cy.xpath("//div[contains(text(), 'Please enter a valid 10-digit phone number.')]");
    }
    getRegisterLinkHomePage() {
        return cy.xpath("//a[contains(text(), 'Signup')]");
    }

    enterFirstName(firstName) {
        if (firstName) this.typeInElement(this.getFirstNameField(), firstName);
    }
    enterLastName(lastName) {
        if (lastName) this.typeInElement(this.getLastNameField(), lastName);
    }
    enterEmail(email) {
        if (email) this.typeInElement(this.getEmailField(), email);
    }
    enterPassword(password) {
        if (password) this.typeInElement(this.getPasswordField(), password);
    }
    enterConfirmPassword(confirmPassword) {
        if (confirmPassword) this.typeInElement(this.getConfirmPasswordField(), confirmPassword);
    }
    enterPhoneNumber(phoneNumber) {
        if (phoneNumber) this.typeInElement(this.getPhoneNumberField(), phoneNumber);
    }

    clickFirstNextButton() {
        this.clickElement(this.getFirstNextButton());
    }
    clickSecondNextButton() {
        this.clickElement(this.getSecondNextButton());
    }
    clickSignupButton() {
        this.clickElement(this.getSignupButton());
    }

    navigate() {
        cy.visit("/");
        this.navigateToJobSeeker();
        this.clickElement(this.getRegisterLinkHomePage());
    }

    register(firstName, lastName, email, password, confirmPassword, phoneNumber) {
        this.enterFirstName(firstName);
        this.enterLastName(lastName);
        this.clickFirstNextButton();
        this.enterEmail(email);
        this.enterPassword(password);
        this.enterConfirmPassword(confirmPassword);
        this.clickSecondNextButton();
        this.enterPhoneNumber(phoneNumber);
        this.clickSignupButton();
    }

    fillFirstRegistrationPage(firstName, lastName) {
        this.enterFirstName(firstName);
        this.enterLastName(lastName);
        this.clickFirstNextButton();
    }

    fillSecondRegistrationPage(firstName, lastName, email, password, confirmPassword) {
        this.fillFirstRegistrationPage(firstName, lastName);
        this.enterEmail(email);
        this.enterPassword(password);
        this.enterConfirmPassword(confirmPassword);
        this.clickSecondNextButton();
    }
}

File: Hunt_Career_TestNG_Hybrid_Framework/src/main/java/com/huntcareer/qa/pages/RegisterPage.java
package com.huntcareer.qa.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class RegisterPage extends PageBase {

    @FindBy(xpath = "//input[@id='firstName']")
    private WebElement firstNameField;

    @FindBy(xpath = "//input[@id='lastName']")
    private WebElement lastNameField;

    @FindBy(xpath = "//input[@id='email']")
    private WebElement emailField;

    @FindBy(xpath = "//input[@id='password']")
    private WebElement passwordField;

    @FindBy(xpath = "//input[@id='confirmPassword']")
    private WebElement confirmPasswordField;

    @FindBy(xpath = "//input[@id='phoneNumber']")
    private WebElement phoneNumberField;

    @FindBy(xpath = "//button[@type='button']")
    private WebElement firstNextButton;

    @FindBy(xpath = "//button[normalize-space()='Next']")
    private WebElement secondNextButton;

    @FindBy(xpath = "//button[@aria-label='Sign up']")
    private WebElement signUpButton;

    @FindBy(xpath = "//div[contains(text(),'User already exists')]")
    public WebElement duplicateErrorMessage;

    @FindBy(xpath = "//div[contains(text(),'First Name is required.')]")
    public WebElement firstNameBlankMessage;

    @FindBy(xpath = "//div[contains(text(),'Last Name is required.')]")
    public WebElement lastNameBlankMessage;

    @FindBy(xpath = "//div[contains(text(),'Email is required.')]")
    public WebElement emailBlankMessage;

    @FindBy(xpath = "//div[contains(text(),'Please enter a valid email address.')]")
    public WebElement pleaseEnterValidEmailAddressMessage;

    @FindBy(xpath = "//div[contains(text(),'Password must be at least 8 characters long.')]")
    public WebElement passwordMustHaveRequiredNumberCharactersMessage;

    @FindBy(xpath = "//div[contains(text(),'Passwords do not match.')]")
    public WebElement passwordNotMatchMessage;

    @FindBy(xpath = "//div[contains(text(),'Phone Number is required.')]")
    public WebElement phoneNumberRequiredMessage;

    @FindBy(xpath = "//div[contains(text(),'Please enter a valid 10-digit phone number.')]")
    public WebElement phoneNumberInvalidMessage;

    public RegisterPage(WebDriver driver) {
        super(driver);
        PageFactory.initElements(driver, this);
    }

    public void enterFirstName(String firstNameText) {
        typeInElement(firstNameField, firstNameText);
    }

    public void enterLastName(String lastNameText) {
        typeInElement(lastNameField, lastNameText);
    }

    public void clickFirstNextButton() {
        clickElement(firstNextButton);
    }

    public void enterEmail(String emailText) {
        typeInElement(emailField, emailText);
    }

    public void enterPassword(String passwordText) {
        typeInElement(passwordField, passwordText);
    }

    public void enterConfirmPassword(String confirmPasswordText) {
        typeInElement(confirmPasswordField, confirmPasswordText);
    }

    public void clickSecondNextButton() {
        clickElement(secondNextButton);
    }

    public void enterPhoneNumber(String phoneNumber) {
        typeInElement(phoneNumberField, phoneNumber);
    }

    public LoginPage clickSignUpButton() {
        clickElement(signUpButton);
        return new LoginPage(driver);
    }
}
